server/datastore/mysql/scheduled_queries_test.go

```diff
@@ -365,3 +369,248 @@ func testScheduledQueriesCascadingDelete(t *testing.T, ds *Datastore) {
 	require.Nil(t, err)
 	require.Len(t, gotQueries, 1)
 }
+
+func testScheduledQueriesIDsByName(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+	user := test.NewUser(t, ds, "User", "user@example.com", true)
+	queries := []*fleet.Query{
+		{Name: "foo", Description: "get the foos", Query: "select * from foo"},
+		{Name: "bar", Description: "do some bars", Query: "select * from bar"},
+		{Name: "foo2", Description: "get the foos", Query: "select * from foo2"},
+		{Name: "bar2", Description: "do some bars", Query: "select * from bar2"},
+	}
+	err := ds.ApplyQueries(ctx, user.ID, queries)
+	require.NoError(t, err)
+
+	specs := []*fleet.PackSpec{
+		{
+			Name:    "baz",
+			Targets: fleet.PackSpecTargets{Labels: []string{}},
+			Queries: []fleet.PackSpecQuery{
+				{
+					QueryName:   queries[0].Name,
+					Description: "test_foo",
+					Interval:    60,
+				},
+				{
+					QueryName:   queries[2].Name,
+					Description: "test_foo2",
+					Interval:    60,
+				},
+			},
+		},
+		{
+			Name:    "qux",
+			Targets: fleet.PackSpecTargets{Labels: []string{}},
+			Queries: []fleet.PackSpecQuery{
+				{
+					QueryName:   queries[1].Name,
+					Description: "test_bar",
+					Interval:    60,
+				},
+				{
+					QueryName:   queries[3].Name,
+					Description: "test_bar2",
+					Interval:    60,
+				},
+			},
+		},
+	}
+	err = ds.ApplyPackSpecs(ctx, specs)
+	require.NoError(t, err)
+
+	// load the scheduled query IDs as that is what we want to test
+	bazPack, _, err := ds.PackByName(ctx, "baz")
+	require.NoError(t, err)
+	quxPack, _, err := ds.PackByName(ctx, "qux")
+	require.NoError(t, err)
+
+	sqsBaz, err := ds.ListScheduledQueriesInPack(ctx, bazPack.ID)
+	require.NoError(t, err)
+	require.Len(t, sqsBaz, 2)
+	sqsQux, err := ds.ListScheduledQueriesInPack(ctx, quxPack.ID)
+	require.NoError(t, err)
+	require.Len(t, sqsQux, 2)
+
+	oldBatchSize := scheduledQueryIDsByNameBatchSize
+	scheduledQueryIDsByNameBatchSize = 2
+	defer func() { scheduledQueryIDsByNameBatchSize = oldBatchSize }()
```

Comment:
Nit: This quirk can be avoided by either adding a `batchSize` argument or an "Option argument" to `ScheduledQueryIDsByName`.

Reply:
Ah yeah, that'd be a bit nicer, I'll update this.